lib/Dialects/VectorExt/Transform/VectorWarpUtils.cpp

```diff
@@ -368,12 +368,25 @@ struct WarpOpForwardOperand : public OpRewritePattern<WarpSingleLaneOp> {
     Value valForwarded;
     unsigned resultIndex;
     for (OpOperand &operand : yield->getOpOperands()) {
+      Value result = warpOp.getResult(operand.getOperandNumber());
+      if (result.use_empty())
+        continue;
+
+      if (auto defOp = operand.get().getDefiningOp()) {
+        if (warpOp->isAncestor(defOp))
+          continue;
+        // Assume all the values coming from above are uniform.
+        if (result.getType() != operand.get().getType())
+          continue;
+        valForwarded = operand.get();
+        resultIndex = operand.getOperandNumber();
+        continue;
+      }
       auto arg = operand.get().dyn_cast<BlockArgument>();
       if (!arg || arg.getOwner()->getParentOp() != warpOp.getOperation())
```

Comment:
Not sure if there's a use case for it, but it would be easy to also support bbArgs from a parent op (not just the warpOp).

Reply:
good, point, change the logic to handle both